### ISSUE TITLE
memory leak improvement for AsyncHttpRequest?

### ISSUE THREAD

**Issue by snowpong** [android-async-http/android-async-http#41]
I ran into a memory leak that might be improved by a one-liner in AsyncHttpRequest::run(...)

When a developers calls AsyncHttpClient::get(...) he passes an AsyncHttpResponseHandler along.
This AsyncHttpResponseHandler is often (I believe) implemented as an internal anonymous class, in for example an Activity.
As it's an inner class it implicitly has a this pointer to it's parent - in this case, the Activity.

This is OK as long as  nobody else is keeping a reference to the instance of this response handler.

However, AsyncHttpRequest keeps a reference to the responseHandler, and AsyncHttpClient keeps a reference to the AsyncHttpRequest in it's thread pool.

So in other words, if you implement  AsyncHttpResponseHandler as an internal class, the parent class is kept alive until the thread pool overwrites your AsyncHttpRequest with a new one. Which could be quite some time.

I suggest you simply reset responseHandler = null at the end of  AsyncHttpRequest::run(...).

You could argue that people shouldn't pass in handlers with implicit parent pointers, and I agree - but then again, there is no need to keep the runnable task around in the pool after it's finished either?
